scripts/base/frameworks/cluster/main.zeek

```diff
@@ -347,7 +347,7 @@ function nodeid_topic(id: string): string
 	return nodeid_topic_prefix + id + "/";
 	}
 
-@if ( Cluster::is_enabled() )
+@if ( Cluster::is_enabled() ) &analyze
```

Comment:
Still think this `@if` should be dropped again and runtime checks are fine.

@J-Gras , do you remember why it was added in  3db8bb4a44e39ca87699060ee1317a871c6342b9?

The one reason I could see is a left-over from not being able to suppress deprecation warnings and putting it in place as a way to suppress them (because many tests don't run with Cluster::is_enabled()), but otherwise it's not obvious to me why this came in.

Reply:
> do you remember why it was added in https://github.com/zeek/zeek/commit/3db8bb4a44e39ca87699060ee1317a871c6342b9?

I think the intention was just to make things a little cleaner by skipping event handlers that won't be used anyway. Seems `Broker::peer_added` was (and is 🙄 ) the only handler that checks for `Cluster::is_enabled()`. For example, `Log::create_stream` is called unconditionally from `zeek_init`.

Reply:
> However I agree that we shouldn't unconditionally create the log stream if it's clear it won't actually be used.  It'd be easy to adapt the zeek_init handler in that way.

Let me chime in here with a long-winded opinion that we should create that stream - this is much unrelated to "@if &analyze", sorry Vern.

* If we can avoid any ifs (static or dynamic) around cluster, IMO things are simpler as there's just one scenario to think about. Adding `@if` or `if` does not make things simpler.
* There won't actually be a cluster.log file created as long as no one `Log::write()`'s to it (I think not event a writer backend instantiated), so the stream shouldn't cost in the non-cluster mode.
* If you do get a `cluster.log` running in a non-cluster `zeek -r` setup, that's curious. A Log::write() happened that maybe shouldn't have. I think you do want to see it.
* But, if we don't create the log stream, a ``Log::write()`` to it is a silent dropped log message. Not great, would rather have that surprising cluster.log in non-cluster mode show up.
* Specifically because there's this ``Cluster::log()`` helper. Likely that's only used when `Cluster::is_enabled() == T`, but it's not obvious that it now won't work in non-cluster mode. ``Cluster::create_master()`` is particularly funky as it caters for the non-cluster case, too, but also has calls to `Cluster::log()`, but in the "right places".

* There's also at least one known schema generator that iterates over all active log streams and appears to have a non-cluster mode (not sure it's used). If we don't create the Cluster::LOG stream, we're hiding it form that tool when used as `zeek generate-log-schema.zeek` (does that fall under API/interface stability/regression? :-) )

So all in all, I think that stream being there even in a non-cluster setup has almost no cost, voids having that extra `@if` / `if`, avoids surprises also has been there in Zeek 5.2.

Reply:
> Adding @if or if does not make things simpler.

Sure, but the cluster code is always loaded, but people aren't always running a cluster. I was thinking about this more earlier — I think the whole structure of that framework is a bit off. It should separate config state that needs to be there for any setup to reason about (as in, are we running a cluster or not?), and separate stuff that only matters when a cluster is actually running (like, a cluster log).

> If we don't create the Cluster::LOG stream, we're hiding it form that tool

Yes, I actually thought that's a plus. 

> and also has been there in Zeek 5.2 and prevoiusly.

Isn't the `Log::create_stream` currently guarded by `Cluster::is_enabled()`?

Anyway — not a hill I'm prepared to die on and I've spent all day reviewing PRs. You have my opinion. :-)

Reply:
So I'm not sure I could follow all of that well enough to know whether I need to make a change to this facet of the PR. For now, I'll leave it as I originally* submitted it, we can circle back if more refinements needed. (* = well this whole PR is itself not "original", but you know what I mean ... :P)

Reply:
> ... know whether I need to make a change to this facet of the PR.

Can leave as is. IMO removing the `@if`'s preferred, but resolving as that can be a follow-up or they'll just linger around.